azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodTemplate.java

```diff
@@ -0,0 +1,140 @@
+   */
+  public V1Pod getPodFromTemplate() {
+    return this.podFromTemplate;
+  }
+
+  /**
+   * @param filterPredicate Predicate to filter the init containers.
+   * @return The list of filtered init Containers derived from the POD specified in the template.
+   */
+  public List<V1Container> getInitContainers(Predicate<? super V1Container> filterPredicate) {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    if (null == spec) {
+      return Collections.emptyList();
+    }
+    List<V1Container> initContainers = spec.getInitContainers();
+    return initContainers == null ? Collections.emptyList() :
+        initContainers.stream().filter(filterPredicate).collect(Collectors.toList());
+  }
+
+  /**
+   * @return The list of init Containers derived from the POD specified in the template.
+   */
+  public List<V1Container> getInitContainers() {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    return spec == null ? Collections.emptyList() : spec.getInitContainers();
+  }
+
+  /**
+   * @param filterPredicate Predicate to filter the volumes.
+   * @return The list of filtered Volumes derived from the POD specified in the template.
+   */
+  public List<V1Volume> getVolumes(Predicate<? super V1Volume> filterPredicate) {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    if (null == spec) {
+      return Collections.emptyList();
+    }
+    List<V1Volume> volumes = spec.getVolumes();
+    return volumes == null ?
+        Collections.emptyList() :
+        volumes.stream().filter(filterPredicate).collect(Collectors.toList());
+  }
+
+  /**
+   * @return The list of filtered Volumes derived from the POD specified in the template.
+   */
+  public List<V1Volume> getVolumes() {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    return spec == null ? Collections.emptyList() : spec.getVolumes();
+  }
+
+  /**
+   * This method returns volume mounts only for the first container.
+   *
+   * @param filterPredicate Predicate to filter the Volume Mounts.
+   * @return The list of filtered Volume Mounts to the appContainer derived from the POD specified
+   * in the template.
+   */
+  public List<V1VolumeMount> getContainerVolumeMounts(
+      Predicate<? super V1VolumeMount> filterPredicate) {
+    List<V1VolumeMount> volumeMounts =
+        this.podFromTemplate.getSpec().getContainers().get(0).getVolumeMounts();
+    return volumeMounts.stream().filter(filterPredicate).collect(Collectors.toList());
+  }
+
+  /**
+   * This method returns volume mounts only for the first container.
+   *
+   * @return The list of Volume Mounts to the appContainer derived from the POD specified in the
+   * template.
+   */
+  public List<V1VolumeMount> getContainerVolumeMounts() {
+    V1PodSpec spec = this.podFromTemplate.getSpec();
+    List<V1Container> containers = spec == null ? Collections.emptyList() : spec.getContainers();
+    return containers.isEmpty() ? Collections.emptyList() :
+        containers.get(0).getVolumeMounts();
```

Comment:
Do see need of another method which gives volumeMounts for any specific container?

Reply:
Volume mounts are needed only for the application container(non-init container). As of the current design, flow-container will be the only non-init container.
Init containers from the template are taken completely and hence their volume mounts are already included.